File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        """
        :type target: int
        :type nums: List[int]
        :rtype: int
        """
        left = 0
        sum_ = 0
        min_len = float("inf")

        for right in range(len(nums)):
            sum_ += nums[right]
            while sum_ >= target:
                min_len = min(min_len, right - left + 1)
                sum_ -= nums[left]
                left += 1

        return min_len if min_len != float("inf") else 0

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0;
        int sum = 0;
        int minLength = Integer.MAX_VALUE;
        for (int right = 0; right < nums.length; right++){
            sum += nums[right];
            while(sum >= target){
                if(right-left+1 < minLength){
                    minLength = right - left + 1;
                }
                sum -= nums[left];
                left++;
            }
        }
        return minLength != Integer.MAX_VALUE?minLength:0;
    }
}